core/includes/config.inc

```diff
@@ -224,6 +227,33 @@ function config_uninstall_config($module, $config_name = NULL) {
 }
 
 /**
+ * Get the storage object for the specified configuration type
+ *
+ * @param string $type
+ *   (optional) The type of config directory to return. Backdrop core provides
+ *   'active' and 'staging'. Defaults to 'active'.
+ *
+ * @return Config
+ *   A ConfigStorageInterface object managing the specified configuration type.
+ */
+function config_get_config_storage($type = 'active') {
+  $directory = config_get_config_directory($type);
+  $colon_pos = strpos($directory, ':');
+  if ($colon_pos !== FALSE) {
+    $storage_type = substr($directory, 0, $colon_pos);
+    switch ($storage_type) {
+      case 'db':
+        return new ConfigDatabaseStorage($directory);
+      case 'file':
+        return new ConfigFileStorage(substr($directory, $colon_pos+1));
+      default:
+        // fall through
```

Comment:
We could take this a bit further and allow additional storage types to be used by pulling from `settings_get()`, similar to how alternative cache back-ends are specified. See the `cache()` function for examples.